plat/mediatek/common/mtk_sip_svc.c

```diff
@@ -56,11 +72,9 @@ static uint64_t mediatek_sip_handler(uint32_t smc_fid,
 		SMC_RET1(handle, ret);
 
 	default:
-		ERROR("%s: unhandled SMC (0x%x)\n", __func__, smc_fid);
-		break;
+		return mediatek_plat_sip_handler(smc_fid, x1, x2, x3, x4,
+			cookie, handle, flags);
```

Comment:
It seems strange to me to provide a hook here from plat/mediatek/common code to the platform specific code, but then not share any other code with the other upstream MTK platform. Not even the mt_sip_set_authorized_sreg() implementation. You might as well provide separate SiP service code (or better, do more code sharing).